webapp/src/ts/effects/contacts.effects.ts

```diff
@@ -116,50 +121,47 @@ export class ContactsEffects {
     );
   },{ dispatch: false });
 
-  updateSelectedContactSummary = createEffect(() => {
+  receiveSelectedContactTargetDoc = createEffect(() => {
     return this.actions$.pipe(
-      ofType(ContactActionList.receiveSelectedContactReports),
-      withLatestFrom(
-        this.store.pipe(select(Selectors.getSelectedContact))
-      ),
-      exhaustMap(([, selectedContact]) => {
-        const selected: any = { ...selectedContact };
-        this.contactsActions.setContactsLoadingSummary(true);
-        this.tasksForContactService
-          .get(selected)
-          .then((tasks) => {
-            return this.contactsActions.updateSelectedContactsTasks(tasks);
-          });
-        return from(this.contactSummaryService.get(selected.doc, selected.reports, selected.lineage)).pipe(
-          map(summary => {
-            this.contactsActions.setContactsLoadingSummary(false);
-            return this.contactsActions.updateSelectedContactSummary(summary);
+      ofType(ContactActionList.loadSelectedContactTargetDoc),
+      exhaustMap(({ payload: { selected } }) => {
+        return from(this.targetAggregateService.getCurrentTargetDoc(selected)).pipe(
+          map(targetDoc => {
+            const selectedContact = { ...selected, targetDoc };
```

Comment:
Like above, I don't think effects should do the work of reducers.   

What I think would makes things clearer is: 
- call `receiveSelectedContactTargetDoc` action with the targetDoc param and have a reducer save that in the store
- call `loadSelectedContactSummary` which doesn't need any payload
- the effect "updateSelectedContactSummary"  should wait for `loadSelectedContactSummary` and get the `selectedContact` from the store and load the contact summary as usual

Reply:
Great suggestion. I have refactored as suggested.